app/src/main/java/com/arygm/quickfix/ui/search/QuickFixFinder.kt

```diff
@@ -0,0 +1,121 @@
+import kotlinx.coroutines.launch
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun QuickFixFinderScreen(
+    navigationActions: NavigationActions,
+    navigationActionsRoot: NavigationActions,
+    isUser: Boolean = true
+) {
+  Scaffold(
+      containerColor = colorScheme.background,
+      topBar = {
+        TopAppBar(
+            title = {
+              Text(
+                  text = "Quickfix",
+                  color = colorScheme.primary,
+                  style = MaterialTheme.typography.headlineLarge,
+                  modifier = Modifier.testTag("QuickFixFinderTopBarTitle"))
+            },
+            colors = TopAppBarDefaults.topAppBarColors(containerColor = colorScheme.background),
+            modifier = Modifier.testTag("QuickFixFinderTopBar"))
+      },
+      content = { padding ->
+        Column(
+            modifier = Modifier.fillMaxSize().testTag("QuickFixFinderContent").padding(padding),
+            verticalArrangement = Arrangement.Center,
+            horizontalAlignment = Alignment.CenterHorizontally) {
+              val pagerState = rememberPagerState(pageCount = { 2 })
+              val coroutineScope = rememberCoroutineScope()
+
+              Surface(
+                  color = colorScheme.surface,
+                  shape = RoundedCornerShape(20.dp),
+                  modifier = Modifier.padding(horizontal = 40.dp).clip(RoundedCornerShape(20.dp))) {
+                    TabRow(
+                        selectedTabIndex = pagerState.currentPage,
+                        containerColor = Color.Transparent,
+                        divider = {},
+                        indicator = {},
+                        modifier =
+                            Modifier.padding(horizontal = 1.dp, vertical = 1.dp)
+                                .align(Alignment.CenterHorizontally)
+                                .testTag("quickFixSearchTabRow")) {
+                          QuickFixScreenTab(pagerState, coroutineScope, 0, "Search")
+                          QuickFixScreenTab(pagerState, coroutineScope, 1, "Announce")
+                        }
+                  }
+
+              HorizontalPager(
+                  state = pagerState,
+                  userScrollEnabled = false,
+                  modifier = Modifier.testTag("quickFixSearchPager")) { page ->
+                    when (page) {
+                      0 -> SearchOnBoarding(navigationActions, navigationActionsRoot, isUser)
+                      1 -> AnnouncementScreen(navigationActions, isUser)
+                      else -> Text("Should never happen !")
+                    }
+                  }
+            }
+      })
+}
+
+@Composable
+fun QuickFixScreenTab(
+    pagerState: PagerState,
+    coroutineScope: CoroutineScope,
+    currentPage: Int,
+    title: String
+) {
+  Tab(
+      selected = pagerState.currentPage == currentPage,
+      onClick = { coroutineScope.launch { pagerState.scrollToPage(currentPage) } },
```

Comment:
Good use of the coroutine scope but you could just use a LaunchedEffect that automatically provides a coroutine scope. It will make the code cleaner but it's not mandatory, just a remark for the future implementations.  
P.S: you should resolve this conversation.